### PR TITLE
Initial commit of a3-megagpu-8g Slurm blueprint

### PR DESCRIPTION
**Pull request by tpdownes**
The initial commit includes the following

- a3-megagpu-8g blueprint following same pattern as a3-highgpu-8g
- intentionally basic instructions; these will be replaced by documentation on cloud.google.com prior to release
- an integration test to test image building

A follow-up PR will test cluster operation as in #2617 

### Submission Checklist

Please take the following actions before submitting this pull request.

* Fork your PR branch from the Toolkit "develop" branch (not main)
* Test all changes with pre-commit in a local branch [#](https://goo.gle/hpc-toolkit#development)
* Confirm that "make tests" passes all tests
* Add or modify unit tests to cover code changes
* Ensure that unit test coverage remains above 80%
* Update all applicable documentation
* Follow Cloud HPC Toolkit Contribution guidelines [#](https://goo.gle/hpc-toolkit-contributing)

### CODE REVIEW

File: examples/machine-learning/a3-megagpu-8g/deployment-base.yaml [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+terraform_backend_defaults:
+  type: gcs
+  configuration:
+    bucket: customer-bucket
+
+vars:
+  project_id: customer-project
+  region: customer-region
+  zone: customer-zone
+  network_name_system: a3mega-sys-net
```
Comment (harshthakkar01) [inline examples/machine-learning/a3-megagpu-8g/deployment-base.yaml:25]: Should we update this hard coded value ?

Comment (tpdownes) [reply examples/machine-learning/a3-megagpu-8g/deployment-base.yaml]: Addressed by 769edc6310e9c14d85fc47c3f9670a16a2817138

File: examples/machine-learning/a3-megagpu-8g/nccl-tests/README [hunk 2]
```diff
@@ -0,0 +1,89 @@
+The examples in this directory are used to show how enroot + pyxis can be used
```
Comment (mr0re1) [inline examples/machine-learning/a3-megagpu-8g/nccl-tests/README:1]: nit `s/README/README.md/` ?

File: examples/machine-learning/a3-megagpu-8g/slurm-a3mega-cluster.yaml [hunk 3]
```diff
@@ -0,0 +1,260 @@
...
+      enable_public_ips: false
+      enable_smt: true
+      node_conf:
+        CoresPerSocket: 52
+        ThreadsPerCore: 2
+      on_host_maintenance: TERMINATE
+      bandwidth_tier: gvnic_enabled
+      reservation_name: $(vars.a3mega_reservation_name)
+      maintenance_interval: $(vars.a3mega_maintenance_interval)
+      startup_script: $(a3mega_startup.startup_script)
```
Comment (mr0re1) [inline examples/machine-learning/a3-megagpu-8g/slurm-a3mega-cluster.yaml:179]: nit. Move to `use`.

File: examples/machine-learning/a3-megagpu-8g/slurm-a3mega-cluster.yaml [hunk 4]
```diff
@@ -0,0 +1,260 @@
...
+    - homefs
+    - data-bucket
+    settings:
+      machine_type: c2-standard-8
+      enable_cleanup_compute: true
+      enable_external_prolog_epilog: true
+      enable_devel: false
+      slurm_conf_tpl: modules/embedded/community/modules/scheduler/schedmd-slurm-gcp-v6-controller/etc/long-prolog-slurm.conf.tpl
+      enable_controller_public_ips: false
+      controller_startup_script: $(controller_startup.startup_script)
```
Comment (mr0re1) [inline examples/machine-learning/a3-megagpu-8g/slurm-a3mega-cluster.yaml:250]: nit. Move to `use`

**Review by mr0re1 — APPROVED** [review]

File: examples/machine-learning/a3-megagpu-8g/slurm-a3mega-cluster.yaml (hunk 3 above)
Comment (tpdownes) [reply examples/machine-learning/a3-megagpu-8g/slurm-a3mega-cluster.yaml:179]: ack for future work.

File: examples/machine-learning/a3-megagpu-8g/slurm-a3mega-cluster.yaml (hunk 4 above)
Comment (tpdownes) [reply examples/machine-learning/a3-megagpu-8g/slurm-a3mega-cluster.yaml:250]: ack for future work.

File: examples/machine-learning/a3-megagpu-8g/nccl-tests/README (hunk 2 above)
Comment (tpdownes) [reply examples/machine-learning/a3-megagpu-8g/nccl-tests/README:1]: ack for future work

**Review by harshthakkar01 — APPROVED** [review]

**Pull request merged**